### DISCUSSION TITLE
Parking-v0 With GrayScale Image Observation: No Good Results

### DISCUSSION THREAD

**Discussion by LainWeigh** [eleurent/highway-env#439] [Q&A]
Hi, all.

I am trying to use the SAC algorithm (without HER) to train the parking-v0 environment under the observation conditions of GrayScale Image.

My observations are as follows:
```python
"observation": {
    "type": "GrayscaleObservation",
    "observation_shape": (540, 320),
    "stack_size": 4,  
    "weights": [0.2989, 0.5870, 0.1140],  # weights for RGB conversio
    "scaling": 7,
},
```

However, no matter what algorithm hyperparameters, network structure, and reward function I use, the car always circles in the parking lot without learning to park.

What is the reason for this? How would I tweak the above to get the parking effect?

Thanks a lot!

**Comment by eleurent** [comment]
Hi,
I'm not sure what is going wrong, I think the observation should contain the relevant features (vehicle state and goal location), and the reward is dense, so SAC should be able to optimize that.
I'll try to TAL when I find the time.

**Reply by LainWeigh** [reply]
Thank you for your kindly answer!

Like the observations used in `DQN-Nature (2015)`, my `observations` are full images of the parking lot stitched from the past 4 frames, and each image has a size of 84×84(I've changed that). The `reward` uses the original reward in highway.

After checking and debugging the code, I found the reason is that **the `CNN` network in sb3 cannot extract the features of the image** (the output of the `CNN` is all 0), so the mlp network cannot be trained. Do you have any experience with feature extraction for such images?

Thanks!

**Comment by eleurent** [comment]
Oh, that is very strange indeed. There must be something wrong in how your CNN model is configured then, because the signal should definitely be present in 84x84 grayscale image observations, and the CNN should definitely be able to learn from it (and not output 0s). Looks like this is an issue on the agent side, so its probably worth asking the SB3 folks directly :)

**Comment by LainWeigh** [comment]
Thanks so much!
